frontend/components/forms/image-uploader/component.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { ChangeEvent, useState } from 'react';
+
+import { FieldValues } from 'react-hook-form';
+import { useIntl } from 'react-intl';
+
+import Image from 'next/image';
+
+import Button from 'components/button';
+import Loading from 'components/loading';
+
+import { ImageUploaderProps } from './types';
+
+export const ImageUploader = <FormValues extends FieldValues>({
+  preview,
+  register,
+  name,
+  id,
+  text,
+  handleChangeImage,
+  registerOptions,
+  ...rest
+}: ImageUploaderProps<FormValues>) => {
+  const { formatMessage } = useIntl();
+  const [imagePreview, setImagePreview] = useState<string>();
+
+  const handleUploadImage = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.currentTarget.files?.length) {
+      const file = e.currentTarget.files[0];
+      const src = URL.createObjectURL(file);
+      setImagePreview(src);
+      handleChangeImage();
+    }
+  };
+
+  return (
+    <div className="flex items-center justify-start">
+      {preview && (
+        <div className="mr-5">
```

Comment:
🟢

Would that be possible that this component receives a `previewClassName` prop for example to customise how the preview is shown? That would make it more reusable.

By default `previewClassName` could have the following props: `mr-5 rounded-full`. For this to work, you'd need to remove the `width`, `height` and `className` props of the `<Image />` component below. You may have to set the `layout` prop as well.